docs/src/pages/quick-start.md

```diff
@@ -26,14 +23,74 @@ function App() {
   return (
     // Provide the client to your App
     <QueryClientProvider client={queryClient}>
-      <Todos />
     </QueryClientProvider>
   )
 }
+```
+
+Here we are creating a client by creating a new instance of the `QueryClient` class.
+A query client can be used to deal with things like cache.
+
+
+Next, we pass the instance to the `QueryClientProvider` which must be a parent to all components that use `react-query`.
+It connects the queryClient with your application. This is quite similar to the `Provider` component that is given by `redux`.
+
+Let's now create a todo component:
+
+```js
+import {
+  useQuery,
+  useQueryClient,
+  QueryClient,
+  QueryClientProvider,
+} from 'react-query'
+import { getTodos, postTodo } from '../my-api'
+
+// ...app component
+
+function Todos() {
+  // Access the client
+  const queryClient = useQueryClient();
+
+  // Queries
+  const query = useQuery('todos', getTodos)
+
+  return (
+    <div>
+      <ul>
+        {query.data.map(todo => (
+          <li key={todo.id}>{todo.title}</li>
+        ))}
+      </ul>
+    </div>
+  )
+}
+
+render(<App />, document.getElementById('root'))
+```
+
+In the `Todo` component, we start by accessing the queryClient we defined earlier using the `useQueryClient` hook.
+A `query` listens to a source of data, in our case, the `getTodos` method.
+The unique key you provide is used internally for re-fetching, caching, and sharing your queries throughout your application.
+
+Queries are generally use with the `GET` HTTP method, mutations are used for HTTP methods like `POST`, `PUT` etc which modify data on the server.
```

Comment:
this is not a requirement and I wouldn't put it on the first page. react-query is promise based, so everything that produces a promise is welcome. It doesn't even have to fetch data over a network at all, and e.g. graphql exclusively uses POST requests for everything ....